schemaregistry-avro/src/main/scala/com/microsoft/azure/schemaregistry/spark/avro/AvroDataToCatalyst.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.azure.schemaregistry.spark.avro
+
+import com.azure.core.util.BinaryData
+import com.azure.core.util.serializer.TypeReference
+import org.apache.avro.generic.GenericRecord
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodeGenerator, CodegenContext, ExprCode}
+import org.apache.spark.sql.catalyst.expressions.{ExpectsInputTypes, Expression, UnaryExpression}
+import org.apache.spark.sql.types._
+
+case class AvroDataToCatalyst(
+     child: Expression,
+     schemaId: String,
+     options: Map[java.lang.String, java.lang.String])
+  extends UnaryExpression with ExpectsInputTypes with Logging{
+
+  override def inputTypes: Seq[BinaryType] = Seq(BinaryType)
+
+  override lazy val dataType: DataType = {
+    val dt = SchemaConverters.toSqlType(schemaRegistryConstructor.expectedSchema).dataType;
+    dt
+  }
+
+  override def nullable: Boolean = true
+
+  @transient private lazy val schemaRegistryConstructor = SchemaRegistryConstructor.init(schemaId, options)
+
+  @transient private lazy val avroConverter = {
+    new AvroDeserializer(schemaRegistryConstructor.expectedSchema, dataType)
+  }
+
+  override def nullSafeEval(input: Any): Any = {
+    val message = new com.azure.core.models.MessageContent()
+      .setBodyAsBinaryData(BinaryData.fromBytes(input.asInstanceOf[Array[Byte]]))
+      .setContentType(("avro/binary+"+schemaRegistryConstructor.schemaId))
+    val genericRecord = schemaRegistryConstructor.serializer.deserialize(message, TypeReference.createInstance(classOf[GenericRecord]))
+    logInfo("Finished deserializing Message Content")
```

Comment:
General comment on logging:
- Can we be more uniform in where we log? Here we are logging after the execution is done but in line 57 you are logging before the line of code is executed. Let's log before the execution to keep things uniform.
- The description should say something along the line of "Deserializing EventData $messageOffset using schema $schemaGuid"
- Can we log the task id? [EventHubUtils](https://github.com/Azure/azure-event-hubs-spark/blob/6efb19211ffed780930d48eb1744ec1863338e01/core/src/main/scala/org/apache/spark/eventhubs/EventHubsUtils.scala#L159) already have an implementation on getting the TaskId.
- This get logged per row in the RDD, so can we be a bit more descriptive so that user get a bit more information? e.g. the generic record being null or not, message id, schema Guid, etc. This helps us debug things when customer capture the logs.

Reply:
fixed